src/google/protobuf/compiler/objectivec/objectivec_map_field.cc

```diff
@@ -170,6 +174,19 @@ void MapFieldGenerator::DetermineForwardDeclarations(
   }
 }
 
+void MapFieldGenerator::DetermineStaticForwardDeclarations(
+    std::set<string>* fwd_decls) const {
+  // Class name is already in "storage_type".
+  const FieldDescriptor* value_descriptor =
+      descriptor_->message_type()->FindFieldByName("value");
+  if (GetObjectiveCType(value_descriptor) == OBJECTIVECTYPE_MESSAGE) {
+    fwd_decls->insert("extern const struct {} " +
+                      value_field_generator_->variable("class_reference") +
+                    " __asm__(\"_OBJC_CLASS_$_\" GPBStringify(" +
```

Comment:
indent is a little funky here.

the value to insert is likely worth a local just to make this a little more readable.

Reply:
Fixed indent. The actual use of this will get cleaned up in the next CL, so for the sake of keeping merges on different branches easy I would prefer to keep this as is for now.

Reply:
This is now fixed.